cmake/thirdparty/FindROCm.cmake

```diff
@@ -0,0 +1,86 @@
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+###############################################################################
+
+find_path(ROCM_PATH
+     NAMES bin/hcc
+     PATHS /opt/rocm
+     DOC "Path to ROCm hcc executable")
+
+
+if(ROCM_PATH)
+    message(STATUS "ROCM_PATH:  ${ROCM_PATH}")
+    set(CMAKE_CXX_COMPILER_ID "HCC")
+
+    set(HSA_PATH "${ROCM_PATH}/hsa")
+
+# this would be the elegant solution, but cmake is not running the command
+# at make time
+#    set(ROCM_COMPILE_FLAGS "`/opt/rocm/hcc/bin/hcc-config  --cxxflags`")
+#    set(ROCM_LINK_FLAGS "`/opt/rocm/hcc/bin/hcc-config  --ldflags`")
+# so we run the command at cmake time
+    execute_process(COMMAND /opt/rocm/hcc/bin/hcc-config --cxxflags OUTPUT_VARIABLE ROCM_COMPILE_FLAGS)
+    execute_process(COMMAND /opt/rocm/hcc/bin/hcc-config --ldflags OUTPUT_VARIABLE ROCM_LINK_FLAGS)
+    set(ROCM_INCLUDE_PATH "${ROCM_PATH}/hcc/include")
+    set(ROCM_ARCH_FLAG "-amdgpu-target=${BLT_ROCM_ARCH}")
+    set(ROCM_LIBRARIES "-lhc_am")
+
+#    set(ROCM_C_COMPILE_FLAGS ${ROCM_COMPILE_FLAGS})
+#    set(ROCM_C_INCLUDE_PATH "${ROCM_PATH}/hcc/include")
+#    set(ROCM_C_LIBRARIES ${ROCM_LIBRARIES})
+#    set(ROCM_C_LIBRARY_PATH "${ROCM_PATH}/hcc/lib")
+#    set(ROCM_C_LINK_FLAGS "${ROCM_C_LIBRARIES} -L${ROCM_C_LIBRARY_PATH} ${ROCM_ARCH_FLAG}")
+
+    set(ROCM_CXX_COMPILE_FLAGS "${ROCM_COMPILE_FLAGS} -Wno-unused-command-line-argument")
+    set(ROCM_CXX_INCLUDE_PATH "${ROCM_PATH}/hcc/include")
+    set(ROCM_CXX_LIBRARIES ${ROCM_LIBRARIES})
+    set(ROCM_CXX_LIBRARY_PATH "${ROCM_PATH}/hcc/lib")
+    set(ROCM_CXX_LINK_FLAGS "${ROCM_LINK_FLAGS} ${ROCM_ARCH_FLAG}")
+
+
+    set(HCC_COMPILER ${ROCM_PATH}/bin/hcc)
+
+
+    set(CMAKE_CXX_INCLUDE_PATH ${ROCM_CXX_INCLUDE_PATH})
+    set(CMAKE_CXX_LINK_FLAGS ${ROCM_CXX_LINK_FLAGS})
+
+    set(CMAKE_CXX_COMPILER "${ROCM_PATH}/bin/hcc" CACHE FILEPATH "HCC compiler" FORCE)
```

Comment:
I can not find an existing host-config example in the blt tree.  I would call this hcc-config.cmake, and I'm guessing it goes in cmake/thirdparty.  Then the setup command is cmake -C cmake/thirdparty/hcc-config.cmake.  
Is this run as a separate command? Or is it included in the main cmake command which in my case is something like:
 cmake -DENABLE_ROCM=ON -DBLT_SOURCE_DIR=/home/chipf/blt /home/chipf/blt/docs/tutorial/blank_project

My RAJA incantation looks like:
cmake -D ENABLE_ROCM=ON -D ENABLE_OPENMP=OFF /home/chipf/raja  and this is after I have pulled the BLT feature/rocm branch.

Reply:
Ahhh.   Just found host-configs in the RAJA project.  That helps clear things up a bit.  I will create one for rocm and include it in my RAJA commits.

Reply:
the blt ones actually live here:

https://github.com/LLNL/blt/tree/master/cmake/blt-test/host-configs

and there are some additional ones in the tutorial:

https://github.com/LLNL/blt/tree/master/docs/tutorial/host-configs

By default CMake will find whatever compiler is in your path, that might be why it reports gcc. 

To avoid surprises, it would be best to pass hcc to cmake if possible (via `CMAKE_CXX_COMPILER` and/or `CMAKE_C_COMPILER`), vs substituting hcc when an option is on.